BJS2-97522-url-shortener-service BJS2-97527 BJS2-97528 BJS2-97530 BJS2-97531 BJS2-97532 BJS2-97533 BJS2-97535 BJS2-97539 BJS2-97542

KuaNySH88 opened the pull request.

KuaNySH88 committed:
BJS2-97522-url-shortener-service
BJS2-97527
BJS2-97528
BJS2-97530
BJS2-97531
BJS2-97532
BJS2-97533
BJS2-97535
BJS2-97539
BJS2-97542

SnurnicynKonstantin commented on src/main/java/faang/school/urlshortenerservice/repository/UrlCacheRepository.java:
```diff
@@ -0,0 +1,31 @@
+        stringRedisTemplate.opsForValue().set(
+                urlCache + ":" + hash,
+                url,
+                7,
```
7 - за что отвечает эта магическая цифра? Может ее в константу поместишь?

KuaNySH88 replied:
перенес в application.yaml

SnurnicynKonstantin commented on src/main/java/faang/school/urlshortenerservice/config/executor/TaskExecutor.java:
```diff
@@ -0,0 +1,27 @@
+@Configuration
+@EnableAsync
+@RequiredArgsConstructor
+public class TaskExecutor {
```
Это конфигурационный файл. Его надо в config перенести

KuaNySH88 replied:
перенес

SnurnicynKonstantin commented on src/main/java/faang/school/urlshortenerservice/controller/UrlShortenerController.java:
```diff
@@ -0,0 +1,51 @@
+    private final UrlShortenerService urlShortenerService;
+
+    @PostMapping
+    public ResponseEntity<CreateUrlResponseDto> createShortUrl(
```
В ответе этого ендпоинта пришло:
```
{
    "shortUrl": "2C"
}
```
Было бы лучше, если в ответе приел бы готовый url, который можно было бы скопировать и вставить в браузер так, что бы сразу перейти на нужную страницу.
Например в ответе можно отдавать - `http://127.0.0.1:8080/api/v1/url/2C`

И ради упрощения советую отказаться от api/v1/. createShortUrl оставить по `/url`, а getOriginalUrl оставить только `/{hash}` (без /url перед ним)

KuaNySH88 replied:
добавил полный url, поправил ендпойнты

SnurnicynKonstantin commented on src/main/java/faang/school/urlshortenerservice/config/context/UserHeaderFilter.java:
```diff
@@ -20,8 +24,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
             userContext.setUserId(Long.parseLong(userId));
-        }else {
-            throw new IllegalArgumentException("Missing required header 'x-user-id'. Please include 'x-user-id' header with a valid user ID in your request.");
+        } else {
```
А ты используешь x-user-id? Если нет, то можно его и удалит что бы в браузере не запрашивалось его

KuaNySH88 replied:
убрал

KuaNySH88 committed:
BJS2-97522-url-shortener-service
BJS2-97527
BJS2-97528
BJS2-97530
BJS2-97531
BJS2-97532
BJS2-97533
BJS2-97535
BJS2-97539
BJS2-97542
Fix review comments

KuaNySH88 committed:
BJS2-97522-url-shortener-service
BJS2-97527
BJS2-97528
BJS2-97530
BJS2-97531
BJS2-97532
BJS2-97533
BJS2-97535
BJS2-97539
BJS2-97542
Fix review comments

SnurnicynKonstantin reviewed (approved):
Отличная работа. 
В мастер мержить не буду, пусть в ветке код остается. Задачка сдана.
